### DISCUSSION TITLE
Is it possible to schedule a one-time task or reschedule if it already exists?

### DISCUSSION THREAD

**Discussion by JsHater** [kagkarlsson/db-scheduler#439] [Q&A] [answered]
Hi!

I would like to schedule a one-time task or reschedule it if it already exists. However, I don't see a straightforward way to accomplish this. The best approach I have found is to do it manually: attempt to locate the execution by task name and ID. If it's present, try to reschedule it; if not, create a new task instance.  

Is it a good idea to add an additional method that performs the functionality described in the current discussion? Or, I can solve my task using existing api.

**Comment by kagkarlsson** [comment, marked as answer]
For now I think you have to use the approach you found. Check for its existance and if present reschedule instead.
